Libraries/LibC/unistd.cpp

```diff
@@ -552,13 +552,13 @@ int set_process_icon(int icon_id)
 
 char* getlogin()
 {
-    static char __getlogin_buffer[256];
-    if (auto* passwd = getpwuid(getuid())) {
-        strlcpy(__getlogin_buffer, passwd->pw_name, sizeof(__getlogin_buffer));
+    static String buffer;
+    if (buffer.is_null()) {
+        if (auto* passwd = getpwuid(getuid())) {
+            buffer = String(passwd->pw_name);
+        }
         endpwent();
-        return __getlogin_buffer;
```

Comment:
Before, this function "usually" returned a string.
Now, this function *always* returns nullptr.

Could it be there's a missing `return buffer.characters();` or something?

Reply:
Yep, thanks for spotting it!